cwiczenie3/osoba.php

```diff
@@ -38,27 +39,27 @@ public function roznicaWieku(Osoba $osoba)
     {
         $Y1 = $osoba->rok;
         $Y2 = $this->rok;
-        $wiek1= new DateTime($Y1);
+        $wiek1 = new DateTime($Y1);
         $wiek2 = new DateTime($Y2);
         $roznica = $wiek1->diff($wiek2);
 
-        switch ($roznica->format('%y'))
-        {
-            case (1);
-                return "Roznica wieku miedzy przyjaciolmi wynosi " . $roznica->format('%y') . " rok.";
-                break;
 
-            case (2) || (3) || (4);
-                return "Roznica wieku miedzy przyjaciolmi wynosi " . $roznica->format('%y') . " lata.";
-                break;
-        }
-        if (($roznica->format('%y'))>(5))
-        {
-            return "Roznica wieku miedzy przyjaciolmi wynosi " . $roznica->format('%y') . " lat.";
-        }
+        return "Roznica wieku miedzy przyjaciolmi wynosi " . $roznica->format('%y') . " lat.";
+    }
 
 
+    public function __set($co, $wartosc)
```

Comment:
Ta metoda pozwala ustawiać Ci każdą zmienną - jest to antypattern. Nigdy nie używaj __set i __get. Zmień to na setter/getter wyspecjalizowany do ustawiania, pobierania transportu.